src/details/ArborX_DetailsVector.hpp

```diff
@@ -0,0 +1,93 @@
+/****************************************************************************
+ * Copyright (c) 2017-2022 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+#ifndef ARBORX_DETAILS_VECTOR_HPP
+#define ARBORX_DETAILS_VECTOR_HPP
+
+#include <ArborX_HyperPoint.hpp>
+
+namespace ArborX::Details
+{
+
+// Same as Point, but a different type (can't inherit from Point as it would
+// run into constexpr and inability to aggregate initialize)
+template <int DIM, typename Coordinate = float>
+struct Vector
+{
+  static_assert(DIM > 0);
+  // Initialization is needed to be able to use Point in constexpr
+  // TODO: do we want to actually want to zero initialize it? Seems like
+  // unnecessary work.
```

Comment:
Stolen from HyperPoint. 🙂 
```suggestion
  // Initialization is needed to be able to use Vector in constexpr
  // TODO: do we want to actually want to zero initialize it? Seems like
  // unnecessary work.
```